### PR TITLE
Documentation: check for invalid references in CI

### PR DESCRIPTION
**Pull request by balsoft**
Documentation: check for invalid references in CI

Uses https://github.com/serokell/xrefcheck to check for invalid references on CI.

Also, fix all the present invalid references to make sure the new job passes.

### CODE REVIEW

**Comment by balsoft** [comment]
CI fail appears unrelated, actual added job is here: https://buildkite.com/o-1-labs-2/mina/builds/19955#f9776f60-763a-4060-90f8-58ce6629ef79

**Review by samuelarogbonlo — APPROVED** [review]
Looks good

File: rfcs/0038-rosetta-construction-api.md [hunk 1]
```diff
@@ -77,9 +77,9 @@ Before the derivation step, we need to generate a keypair. We'll use the private
 
 [derivation]: #derivation
 
-Derivation demands that the public key expected as input be a hex-encoded byte-array value. So we'll [add functionality](#marshalkeys) to the [client-sdk](#marshalkeys), the [generate-keypair binary](#marshalkeys), and the offcial [Coda CLI](#marshalkeys) to marshall the `Fq.t * Fq.t` pair (the native representation of an uncompressed public key).
+Derivation demands that the public key expected as input be a hex-encoded byte-array value. So we'll [add functionality](#marshal-keys) to the [client-sdk](#marshal-keys), the [generate-keypair binary](#marshal-keys), and the offcial [Coda CLI](#marshal-keys) to marshall the `Fq.t * Fq.t` pair (the native representation of an uncompressed public key).
```
Comment (psteckler) [inline rfcs/0038-rosetta-construction-api.md:80]: should this be `Mina CLI`?

**Review by psteckler — APPROVED** [review]
Left one comment

**Review by lk86 — APPROVED** [review]
Product ✅

**Review by imeckler — APPROVED** [review]

**Pull request merged**